modules/openapi-generator/src/test/resources/3_0/6150_model_json_inline.yaml

```diff
@@ -0,0 +1,77 @@
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: Test inline response model
+  description: Test inline response model.
+  license:
+    name: MIT
+paths:
+  /foobar:
+    get:
+      operationId: testOperation
+      description: No type property in modelJson of InlineResponse200
+      responses:
+        200:
+          description: InlineResponse200 itself.
+          content:
+            application/json:
+              schema:
+                type: object
+                # OAS3 spec allows to use any value in format field
+                format: unknown
+                properties:
+                  foo:
+                    type: string
+                  bar:
+                    type: string
+    post:
+      operationId: testOperationPost
+      description: No type property in modelJson of InlineResponse200
+      responses:
+        400:
+          description: InlineResponse200 itself.
+          content:
+            application/json:
+              schema:
+                title: FooBarObject
+                type: object
+                # wrong format just to make sure it won't be stripped
+                format: date-time
```

Comment:
Can you explain the test you are exercising by adding the `format` keyword at the object level? In 3.0.3, the `format` keyword can be specified for primitive types, not for object types. See http://spec.openapis.org/oas/v3.0.3#dataTypeFormat:

> Primitives have an optional modifier property: format.

Or did you mean to add the `format` keyword for the foo property?
```yaml
properties:
  foo:
    format: date-time
    type: string
```

Reply:
Yeah, I know that `format` property is for primitives only, but my goal was to keep `format` and `type` property even in broken schemas. In other words if user declares `format` property in `type: object` by some reason he will likely expect it won't be stripped of by generator.

Reply:
Then I recommend that you rewrite the comment. Writing "Wrong format ..." conveys that the value of `format` is incorrect, it should not be `date-time` but some other value. That's not what you were doing here.

```yaml
# In OAS 3.0.3, the 'format' keyword can be used for primitive types, not for object types, as
# specified in http://spec.openapis.org/oas/v3.0.3#dataTypeFormat.
# The `format` keyword is added in this test object type to show the generator retains the format property
# even in broken schemas.
```

Reply:
Note in the JSON schema specification, the `format` keyword can be used for any instance, not just primitive types. http://json-schema.org/draft/2019-09/json-schema-validation.html#rfc.section.7. In the draft 3.1, the `format` section still indicates the `format` keyword applies to primitive types.

The JSON schema spec does not say much about what the `format` keyword would mean for an object type, all defined formats apply to string instances.

Reply:
```yaml
title: FooBarObject
type: object
# object type cannot contain format, but it still should be presented in resolved model
format: date-time
```
Something like that?

Reply:
No because the JSON schema specification explicitly supports the `format` keyword for any model instance type. See https://github.com/OAI/OpenAPI-Specification/issues/2301 I just opened related to this issue.
I think the OAS should be slightly rephrased to clarify `format` applies to all types (including object), not just primitive types. Mike Ralphson seems to agree.

```yaml
# It is legal to use the `format` keyword for object types. The JSON schema specification explicitly allows this.
# Even if in practice most OAS authors use `format` for primitive types, it should still be allowed to use format for object types.
```